### PR TITLE
Public page: Add release on public page

### PR DESCRIPTION
**Pull request by antoine-le-calloch**
* Add the possibility to create a release, to group public source page in it, with default public source options. 
* Adds a page to see all releases, and a page to see one release in detail with all the sources in it
* Create, edit, and delete a release from the public access dialog.
* Override the releases default options when publishing a single source.

### CODE REVIEW

**Review by mcoughlin — APPROVED** [review]
@antoine-le-calloch 

Works great, just one comment:
The first time, for the link URL, I put in one that had a slash in it, which broke the API endpoint. Maybe you can ensure in the validation that there is not a slash.

Otherwise works great.

**Comment by antoine-le-calloch** [comment]
> @antoine-le-calloch
> 
> Works great, just one comment: The first time, for the link URL, I put in one that had a slash in it, which broke the API endpoint. Maybe you can ensure in the validation that there is not a slash.
> 
> Otherwise works great.

Okey i will take a look, because there is a already a function that normally prevents this. So it shouldn't work as expected.

**Comment by mcoughlin** [comment]
@antoine-le-calloch can you resolve the conflicts?

**Comment by antoine-le-calloch** [comment]
> @antoine-le-calloch can you resolve the conflicts?

@mcoughlin Yeah sure, but the PR still in progress, currently try to correct some group behavior.

**Comment by Theodlz** [comment]
@antoine-le-calloch just checking, is this ready for a review? I can't remember where we landed on that.

**Review by mcoughlin — APPROVED** [review]
Feature works great.

File: skyportal/handlers/api/public_pages/public_source_page.py
Comment (Theodlz) [inline skyportal/handlers/api/public_pages/public_source_page.py:127]: same comment here (scalars.first -> scalar)

Comment (Theodlz) [reply skyportal/handlers/api/public_pages/public_source_page.py:127]: Overall that applies all across you methods in this file (and other files).

File: skyportal/handlers/public/release.py
Comment (Theodlz) [inline skyportal/handlers/public/release.py:47]: As discussed, with some good cache invalidation we could consider caching those in the future. Please open an issue once merged. Let's not do that here.

File: skyportal/models/gcn.py
Comment (Theodlz) [inline skyportal/models/gcn.py:277]: any reason you don't want to use `data.get("sources"):` anymore?

File: skyportal/models/public_pages/public_source_page.py
Comment (Theodlz) [inline skyportal/models/public_pages/public_source_page.py:27]: that doesn't cause any issue for public sources not on a group? Just checking, I don't remember what baselayer does in that case.

File: static/js/components/source/source_publish/SourcePublish.jsx
Comment (Theodlz) [inline static/js/components/source/source_publish/SourcePublish.jsx:51]: To encourage people to pick a release, I think we should have the release tab opened by default at least.

**Review by Theodlz — COMMENTED** [review]
Very nice! Some minor comments and questions. Once read and/or addressed, I can review again and merge

File: skyportal/handlers/api/public_pages/public_release.py
Comment (antoine-le-calloch) [reply skyportal/handlers/api/public_pages/public_release.py]: Outdated.

File: skyportal/models/gcn.py
Comment (antoine-le-calloch) [reply skyportal/models/gcn.py]: No reason at all.

File: skyportal/models/public_pages/public_source_page.py
Comment (antoine-le-calloch) [reply skyportal/models/public_pages/public_source_page.py:27]: It was a mistake to change from CustomUserAccessControl to accessible_by_groups_members.

**Review by Theodlz — APPROVED** [review]
LGTM!

**Comment by Theodlz** [comment]
oh btw I had the same issue with the photometry validation here, but I guess you fixed this one in the spectra PR right? Maybe I'm confusing which one worked but yeah I noticed one worked and one didn't this morning.

**Comment by antoine-le-calloch** [comment]
I have found one bug with my socket that only update the user, and i want to disabled delete button when the user have no right. @Theodlz
I let you know when its down.

**Comment by Theodlz** [comment]
> oh btw I had the same issue with the photometry validation here, but I guess you fixed this one in the spectra PR right? Maybe I'm confusing which one worked but yeah I noticed one worked and one didn't this morning.

Could you confirm this @antoine-le-calloch? If so, I will take one last look at the code and we can try to merge this afternoon.

**Comment by antoine-le-calloch** [comment]
> > oh btw I had the same issue with the photometry validation here, but I guess you fixed this one in the spectra PR right? Maybe I'm confusing which one worked but yeah I noticed one worked and one didn't this morning.
> 
> Could you confirm this @antoine-le-calloch? If so, I will take one last look at the code and we can try to merge this afternoon.

I'm not sure to understand the problem here.

**Comment by Theodlz** [comment]
@antoine-le-calloch I merge the spectra PR. You should be able to resolve conflicts now

**Comment by Theodlz** [comment]
@antoine-le-calloch just like the public pages, this is a pretty significant feature, but there are no tests for it whatsoever. Would you mind at least adding some API tests today? I would like this to be tested so that after your internship is done, we know it still works when we make changes.

**Comment by antoine-le-calloch** [comment]
> @antoine-le-calloch just like the public pages, this is a pretty significant feature, but there are no tests for it whatsoever. Would you mind at least adding some API tests today? I would like this to be tested so that after your internship is done, we know it still works when we make changes.

I add the tests in a new PR: 

Public page: Public source pages tests #5160

**Comment by antoine-le-calloch** [comment]
@Theodlz, I just added a small piece of code here to fix an issue discovered during testing that could occur if an incorrect release ID is provided.

**Review by Theodlz — APPROVED** [review]
LGTM 🚀 ! Let's talk about what we want to add on top of these features (or if anything needs to change) while you're visiting at CIT this week.

**Comment by Theodlz** [comment]
@antoine-le-calloch they were some conflicts with main so I just fixed those, will merge once the tests are green.

**Pull request merged**
